apps/testingaccessibility/src/pages/learn/index.tsx

```diff
@@ -62,17 +62,26 @@ const productQuery = groq`*[_type == "product" && productId == $productId][0]{
   }`
 
 export const getServerSideProps: GetServerSideProps = async ({req}) => {
-  const {product, purchases, token} = await getPurchasedProduct(
-    req,
-    productQuery,
-  )
-  const ability = await getAbilityFromToken(token)
+  const session = await getSession({req})
+
+  // TODO: instead of fetching the product we can generate rules on the session
+  //       which may mean we can avoid this async call but it also isn't hurting
+  //       anything right now. The ability isn't being used to make any decisions
+  //       at this point.
+  const {product, purchases} = await getPurchasedProduct(req, productQuery)
+
+  const ability = getCurrentAbility({rules: session?.rules})
 
-  console.log(`can invite team members: ${ability.can('invite', 'Team')}`)
-  console.log(`can view content: ${ability.can('view', 'Content')}`)
+  console.log(
+    `LEARN: can invite team members: ${ability.can('invite', 'Team')}`,
+  )
+  console.log(`LEARN: can view content: ${ability.can('view', 'Content')}`)
 
   console.log(
-    `can view product: ${ability.can('view', subject('Product', product))}`,
+    `LEARN: can view product: ${ability.can(
+      'view',
+      subject('Product', product),
+    )}`,
   )
```

Comment:
we can make decisions about what to render or how to redirect based on abilities as needed